cpp/src/partitioning/partitioning.cu

```diff
@@ -724,6 +724,35 @@ struct dispatch_map_type {
 
 namespace detail {
 namespace {
+
+/**
+ * @brief This hash function simply returns the input value cast to the
+ * result_type of the functor.
+ */
+template <typename Key>
+struct IdentityHash {
+  using result_type = uint32_t;
+  IdentityHash()    = default;
```

Comment:
Probably because of the `m_seed` but since that has an initial value it should be possible. 
Is there some benefit to declare the default constructor is `constexpr` ?

Reply:
This question came up because other constructor and member functions are constexpr, and the default constructor is not.

Reply:
`m_seed` may not be needed, since it's not used. can `m_seed` be removed?